### ISSUE TITLE
plotProfile() should deparse `xtype` if it is a vector

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2047]
At the moment, using
```R
library(oce)
data(ctd)
plotProfile(ctd, xtype=ctd[["SA"]]/10)
```
yields a plot with no label on the top axis.  In a discussion with @richardsc I suggested that maybe I ought to use `deparse1(substitute(xtype)` in cases where `xtype` is a numeric vector, and to write that as the name of the axis.  (In the above case, the label would become  `ctd[["SA"]]/10`.)

I was about to write this (it's just 2 lines of code) but then I thought that some users might not realize that they can use `xlab` to write a name for that axis, and might have code to write something for a name.  If I change `plotProfile()` to infer an axis name if `xlab=NULL` (which is the default) then such user code would not yield the expected results.

So I am not sure if I ought to make the change.  I've looked at the docs, and they don't say anything about what happens with `xtype` a vector.  So I could see an argument to make the change.

@richardsc and @clayton33 -- what do you think?  Should I change oce so that e.g.
```R
plotProfile(ctd, xtype=ctd[["SA"]]/10)
```
have the top axis labelled with `ctd[["SA"]]/10`?

I propose a +- vote, in which all votes are real numbers from -1 to 1, so that we can make a decision simply by adding the numbers and seeing if positive.

I vote **0.3** for the change, meaning that I am inclined to do it, but not convinced it is the best thing.

Since a CRAN release is coming up in a few weeks, let's close the voting at 1700h this Friday, March 3.

**Comment by richardsc** [comment]
Considering I didn't even realize you could just pass `xtype` as a vector of anything you want (I know, RTFM), I think this is a safe (and good) change. Handy for doing quick plots and still knowing what you've plotted. 

+0.8

**Comment by dankelley** [comment]
The sum so far is 0.3+0.8+0.5 which is non-negative, so I'll make the change in the "develop" branch.  Thanks for the quick votes!

PS. anybody who has old code like
```R
plotProfile(...)
mtext(...) # add a label at top
```
can just add `ylab=""` to the `plotProfile()` call and the rest will be as it was.  So, not much loss.

**Comment by richardsc** [comment]
FYI my solution for this previously would have been something like:
```r
ctd <- oceSetData(ctd, 'newField', ctd[['SA']]/10)
plotProfile(ctd, xtype='newField')
```
rather than to use `mtext` etc.

**Comment by dankelley** [comment]
I have added this in "develop" commit 0923ab43a466ae4f1a01559168853cb4bfde835f. 

Below I'm pasting a snippet from the updated docs (which indicate the change in how it works), plus a reprex that I think indicates all is okay.

If anybody adds a comment saying this is okay, I'll close the issue.  Thanks!

 ----

<img width="626" alt="Screenshot 2023-02-28 at 10 55 52 AM" src="https://user-images.githubusercontent.com/99469/221891120-1bab1b88-5d94-4ee4-946a-55469a8e9de7.png">

----

``` r
# auto-create x label for plotProfile() with xtype=vector
# https://github.com/dankelley/oce/issues/2047
library(oce)
#> Loading required package: gsw
data(ctd)
par(mfrow=c(1, 3))
plotProfile(ctd, xtype=ctd[["SA"]]+10)
plotProfile(ctd, xtype=ctd[["SA"]]+10, xlab="Absolute Salinity plus 10")
plotProfile(ctd, xtype="SA")
```

![](https://i.imgur.com/fYFD5yi.png)

<sup>Created on 2023-02-28 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by richardsc** [comment]
Why does the y-label "Pressure" look bolded in the third plot?

**Comment by dankelley** [comment]
Good Q. I'll look after the seminar.

**Comment by dankelley** [comment]
The bold-face part is very strange.  Part of my problem is being able to actually see the differences.  I am attaching a zip file that has some test code, along with 3 files that it makes: a PDF, a low-res PNG, and a high-res PNG.

My eyes are not quite capable of seeing the bold-face "Pressure" label in the PDF or in the high-res PNG.  I do, however, see it in the low-res PNG.

@richardsc can you look at the 3 plot files to see if you agree?  This is basically an eye test.

If, indeed, the problem is only in the low-res PNG then I don't see how it can be an oce problem.  (Thinking about the code, I also don't see how it could be a problem.)

Another test might be to try to run the code on a non-mac machine.

This is quite confusing.  (PS. I will be at Dal tomorrow.)

[Archive.zip](https://github.com/dankelley/oce/files/10852609/Archive.zip)

**Comment by dankelley** [comment]
Oh, how interesting.  I think I see why.  I am doing something like
```R
plot(..., ylab="SOMETHING", axes=FALSE) 
mtext("SOMETHING", side=2, ...)
```
and I think they are at very slightly different positions from the axis line.  So, at high-res (or PDF) it is hard to see but at low-res, where I am guessing we are talking about near-pixels differences in the x location, the effect comes out as bold facing.

A simple test shows that if I just use `ylab=""` in the `plot()` call, the problem goes away.

So, I'll check into whether I'm doing this for all plot types and fix them all.

@richardsc -- you have great eyes!!!  Thanks a lot for noticing this.  I seldom make PNGs that are small, like those used in reprex examples (esp. in a case like this, with 4 panels so a small font)

**Comment by richardsc** [comment]
Yeah, I wondered if it was an overplotting of text -- I've seen that before when I've mistakenly overwritten something that was already there.

**Comment by dankelley** [comment]
I'll make a checklist of the various `xtype` cases (which is how the code is organized). I will add to this comment as I do blocks of code.

* [ ] `xtype` is a vector (`R/ctd.R` near line 5453: two cases)
* [ ] `

**Comment by dankelley** [comment]
The "bold-faced y label" problem should be solved in "develop" commit f56564b3ddbdd598a1b72802a3a348e7cc25cdd2. 

The code at https://github.com/dankelley/oce/issues/2047#issuecomment-1448335583 (which is now also in `sandbox/issues/20xx/2047`) seems to work fine for me now.  We have:

1. auto-labelling of the x axis
2. a solution to the bold-face y axis name

The edits I just made span over 700 lines of code so I might have introduced problems, but I propose to close this issue at the end of today.  If there are problems, they will crop up before a CRAN release in a month or so.
